### PR TITLE
Fix motion leak

### PR DESCRIPTION
**Pull request by larryTheCoder**

### CODE REVIEW

**Comment by NycuRO** [comment]
👍

**Comment by larryTheCoder** [comment]
Hopefully that this will merged fast as possible

**Comment by vzlomka76** [comment]
@Snake1999 It works well, merge?

**Comment by shanwer** [comment]
@PeratX

File: src/main/java/cn/nukkit/entity/Entity.java [hunk 1]
```diff
@@ -1369,8 +1369,8 @@ public void applyEntityCollision(Entity entity) {
                 dx *= 1.0F + entityCollisionReduction;
                 dz *= 1.0F + entityCollisionReduction;
                 if (this.riding == null) {
-                    motionX = -dx;
-                    motionZ = -dy;
+                    motionX += -dx;
+                    motionZ += -dy;
```
Comment (SupremeMortal) [inline src/main/java/cn/nukkit/entity/Entity.java]: Surely it would make more sense to do `motionX -= dx;`

Comment (larryTheCoder) [inline src/main/java/cn/nukkit/entity/Entity.java]: i guess you right

**Comment by NycuRO** [comment]
#1853

**Comment by PeratX** [comment]
Seems correctly. Need more tests.

**Comment by larryTheCoder** [comment]
@PeratX 
Why don't you tell me at the first time when @shanwer mention you... Nevermind...
This is the tests:
Code:
```java
Player p = Server.getInstance().getPlayer("ThisYourName");
p.setMotion(new Vector3(1, 1, 1));
getServer().getLogger().debug("Player Velocity Treeshold: ");
getServer().getLogger().debug("X: " + p.motionX);
getServer().getLogger().debug("Y: " + p.motionY);
getServer().getLogger().debug("Z: " + p.motionZ);
```
Picture on commit 70a7b5c745d24020b224fdff972eadcb17c8e0d8
bandicam 2017-08-20 22-00-09-179

Picture on this commit
bandicam 2017-08-20 22-08-15-015

You can test it by yourself

**Comment by NycuRO** [comment]
@larryTheCoder Them is shit guy. Don't like anymore. He created one project with name Genisys , abandonate them because don't give donations and now sell project on site chinese.

**Comment by liach** [comment]
What @NycuRO? Do you mean the managers sold this project?

**Comment by PeratX** [comment]
It'a joke https://github.com/NebzzTeam

**Pull request merged**